src/main/java/fr/inria/astor/core/faultlocalization/cocospoon/CocoSpoonEngineFaultLocalizer.java

```diff
@@ -71,35 +63,74 @@ public void runTests(String[] testClasses, ClassLoader cl, ProjectRepairFacade p
 		this.buildTestResultPerSourceLocation(resultsPerNameOfTest, linesExecutedPerTestNames);
 	}
 
-	private void sortBySuspiciousness() {
-		for (SourceLocation sourceLocation : this.countPerSourceLocation.keySet()) {
-			StatementSourceLocation current = new StatementSourceLocation(this.metric, sourceLocation);
-			int ef = 0;
-			int ep = 0;
-			for (TestResult results : this.countPerSourceLocation.get(sourceLocation)) {
-				if (results.isSuccessful())
-					ep++;
-				else
-					ef++;
-			}
-			current.setNf(nbFailingTest - ef);
-			current.setNp(nbSucceedTest - ep);
-			current.setEp(ep);
-			current.setEf(ef);
-			statements.add(current);
+	private MetricParams getMetricParams(SourceLocation sourceLocation) {
+		int ef = 0;
+		int ep = 0;
+		for (TestResult results : this.countPerSourceLocation.get(sourceLocation)) {
+			if (results.isSuccessful())
+				ep++;
+			else
+				ef++;
 		}
-		Collections.sort(statements, new Comparator<StatementSourceLocation>() {
-			@Override
-			public int compare(StatementSourceLocation o1, StatementSourceLocation o2) {
-				return Double.compare(o2.getSuspiciousness(), o1.getSuspiciousness());
-			}
-		});
+
+		return new MetricParams(
+				sourceLocation, ef, ep,
+				nbFailingTest - ef,
+				nbSucceedTest - ep
+		);
+	}
+
+	List<StatementSourceLocation> getStatementsSortedBySuspiciousness() {
+
+        List<StatementSourceLocation> aggregatedResult =
+                countPerSourceLocation.keySet().stream()
+                        .map(this::getMetricParams)
+                        .map(params -> metrics.stream()
+                                .map(metric -> {
+                                    StatementSourceLocation current =
+                                            new StatementSourceLocation(
+                                                    metric,
+                                                    params.getSourceLocation()
+                                            );
+
+                                    current.setEf(params.getEf());
+                                    current.setEp(params.getEp());
+                                    current.setNf(params.getNf());
+                                    current.setNp(params.getNp());
+
+                                    return current;
+                                }).collect(Collectors.toList())
+                        )
+                        .filter(list -> !list.isEmpty())
+                        .map(params -> {
+                            /* TODO: Hier den Vektor normalisieren:
```

Comment:
Hier ist das `TODO`:
- Params ist hier der Vector der Ergebnisse, wobei jeder Eintrag im Vektor für eine der Metriken steht.
- Also wenn wir 3 Metriken haben, ist hier der Vektor der Länge 3
- Am besten speicherst du den Wert in `result`, der Rest sollte dann automatisch klappen

Danach könnte man noch den Spaghetti-Code ein bisschen mehr aufräumen, aber es sollte jedenfalls so schonmal laufen